src/components/responder.js

```diff
@@ -39,11 +39,10 @@ module.exports = class Responder extends Configurable(Component) {
     on(type, listener) {
         super.on(type, (...args) => {
             const rv = listener(...args);
+            const cb = args.pop();
 
-            if (rv && typeof rv.then == 'function') {
-                const cb = args.pop();
-                rv.then((val) => cb(null, val)).catch(cb);
-            }
+            if (!rv || typeof rv.then != 'function' || typeof cb != 'function') return;
```

Comment:
this line is fairly convoluted. can you do multiple returns? something like this if it makes sense:

```js
if (typeof cb != 'function') return
if (!(rv && typeof rv.then == 'function')) return

rv.then((val) => cb(null, val), cb);
```

Reply:
Agreed will change it, with the test case discussion resolution.